packages/headless/src/features/search/search-request.test.ts

```diff
@@ -397,81 +554,167 @@ describe('search request', () => {
   it('#searchRequestParams returns the freezeFacetOrder in state', async () => {
     state.facetOptions = buildMockFacetOptions({freezeFacetOrder: true});
 
-    const params = (await buildSearchRequest(state)).request;
+    const params = (
+      await buildSearchRequest(
+        state,
+        {location: '', referrer: '', userAgent: ''},
+        mockRelay()
+      )
+    ).request;
     expect(params.facetOptions).toEqual({
       freezeFacetOrder: state.facetOptions.freezeFacetOrder,
     });
   });
 
   it('should send visitorId if analytics is enable', async () => {
     state.configuration.analytics.enabled = true;
-    expect((await buildSearchRequest(state)).request.visitorId).toBeDefined();
+    expect(
+      (
+        await buildSearchRequest(
+          state,
+          {location: '', referrer: '', userAgent: ''},
+          mockRelay()
+        )
+      ).request.visitorId
+    ).toBeDefined();
   });
 
   it('should not send visitorId if analytics is disabled', async () => {
     state.configuration.analytics.enabled = false;
     expect(
-      (await buildSearchRequest(state)).request.visitorId
+      (
+        await buildSearchRequest(
+          state,
+          {location: '', referrer: '', userAgent: ''},
+          mockRelay()
+        )
+      ).request.visitorId
     ).not.toBeDefined();
   });
 
   it('#searchRequest.tab holds the #originLevel2', async () => {
     const originLevel2 = 'youtube';
     state.configuration.analytics.originLevel2 = originLevel2;
-    expect((await buildSearchRequest(state)).request.tab).toBe(originLevel2);
+    expect(
+      (
+        await buildSearchRequest(
+          state,
+          {location: '', referrer: '', userAgent: ''},
+          mockRelay()
+        )
+      ).request.tab
+    ).toBe(originLevel2);
   });
 
   it('#searchRequest.referrer holds the #originLevel3', async () => {
```

Comment:
I'm surprised that this test is passing!

Shouldn't it take the referrer from the NavigatorContext passed as a second argument (i.e., '' in this case)?

Reply:
Not with analyticsMode: legacy ;)